src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -0,0 +1,192 @@
+
+import { StripeElement, StripeElements, StripeElementsOptions, StripeUPEClient } from './stripe-upe';
+import StripeUPEPaymentInitializeOptions from './stripe-upe-initialize-options';
+import StripeUPEScriptLoader from './stripe-upe-script-loader';
+
+export default class StripeUPEPaymentStrategy implements PaymentStrategy {
+    private _initializeOptions?: StripeUPEPaymentInitializeOptions;
+    private _stripeUPEClient?: StripeUPEClient;
+    private _stripeElements?: StripeElements;
+    private _stripeElement?: StripeElement;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _stripeScriptLoader: StripeUPEScriptLoader,
+        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _locale: string
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { stripeupe, methodId, gatewayId } = options;
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError('Unable to initialize payment because "gatewayId" argument is not provided.');
+        }
+
+        this._getInitializeOptions(stripeupe);
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(`${gatewayId}?method=${methodId}`));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { initializationData: { stripePublishableKey, stripeConnectedAccount } } = paymentMethod;
+
+        await this._getStripeJs(stripePublishableKey, stripeConnectedAccount);
+        this._stripeElement = await this._mountCardFields(paymentMethod.clientToken);
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+
+        if (!payment || !payment.paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const { paymentData, methodId } = payment;
+        const { shouldSaveInstrument = false, shouldSetAsDefaultInstrument = false } = isHostedInstrumentLike(paymentData) ? paymentData : {};
+
+        const { isStoreCreditApplied : useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+
+        if (useStoreCredit) {
+            await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        }
+
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { clientToken } = paymentMethod;
+        if (!clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
```

Comment:
`clientToken` doesn't seem to be used in this method, why do we care if it is there?